### PR TITLE
Spacewalk Upgrade to Polkadot version 0.9.42

### PR DESCRIPTION
**Pull request by gianfra-t**
This PR updates the Spacewalk dependencies to 0.9.42 and adds changes to correspondingly to the code. This is related to [this issue](https://github.com/pendulum-chain/tasks/issues/158).

Some notable changes can be found in `testchain/node/src/service.rs` where the `WasmExecutor` had changed in between version, and the 4 new [parameters](https://github.com/pendulum-chain/spacewalk/pull/491/files#diff-6053821b8c0c8ae2d671cde8654bbb48f4b5a504996a190cf3241968cc5f321eR129) required for the pallet balances.

Also,`substrate-stellar-sdk` was required to be updated, which is done in branch [polkadot-v0.9.42](https://github.com/pendulum-chain/substrate-stellar-sdk/tree/polkadot-v0.9.42).

### CODE REVIEW

**Review by ebma — APPROVED** [review]
Looks good to me 👍 I just resolved a conflict and fixed some clippy stuff.

File: pallets/oracle/rpc/runtime-api/src/lib.rs [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 use codec::{Codec, Decode, Encode};
-use frame_support::dispatch::DispatchError;
+use frame_support::{dispatch::DispatchError};
 use primitives::UnsignedFixedPoint;
 #[cfg(feature = "std")]
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
-
-#[derive(Eq, PartialEq, Encode, Decode, Default)]
+use scale_info::TypeInfo;
+#[derive(Eq, PartialEq, Encode, Decode, Default, TypeInfo)]
```
[the remark this answers is not in the document]
Comment (gianfra-t) [reply pallets/oracle/rpc/runtime-api/src/lib.rs:10]: `BalanceWrapper<T>` must implement the trait. Must be some new bound introduced by the update.

**Comment by b-yap** [comment]
rerunning the CI,

**Comment by b-yap** [comment]
https://github.com/pendulum-chain/spacewalk/pull/495 has to be closed BEFORE CLOSING this.

**Comment by ebma** [comment]
I'll merge this without CI passing because the job probably failed because our SatoshiPay Stellar archives are currently broken.

**Pull request merged**